restli-tools/src/main/java/com/linkedin/restli/tools/snapshot/gen/PegasusSchemaSnapshotGenerationCmdLineApp.java

```diff
@@ -0,0 +1,114 @@
+/*
+   Copyright (c) 2020 LinkedIn Corp.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+package com.linkedin.restli.tools.snapshot.gen;
+
+import com.linkedin.restli.internal.tools.RestLiToolsUtils;
+import java.io.File;
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.CommandLineParser;
+import org.apache.commons.cli.GnuParser;
+import org.apache.commons.cli.HelpFormatter;
+import org.apache.commons.cli.OptionBuilder;
+import org.apache.commons.cli.Options;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * Tool that encodes pegasus schemas to pegasusSchemaSnapshot files
+ *
+ * @author Yingjie Bi
+ */
+public class PegasusSchemaSnapshotGenerationCmdLineApp
+{
+  private static final Logger _logger = LoggerFactory.getLogger(
+      PegasusSchemaSnapshotGenerationCmdLineApp.class);
+
+  private static final Options _options = new Options();
+
+  static
+  {
+    _options.addOption(OptionBuilder.withLongOpt("help")
+        .withDescription("Print help")
+        .create('h'));
+  }
+
+  public static void main(String[] args) throws Exception
+  {
+    final CommandLineParser parser = new GnuParser();
+    CommandLine cl = parser.parse(_options, args);
+
+    if (cl.hasOption('h'))
+    {
+      help();
+      System.exit(0);
+    }
+
+    String[] cliArgs = cl.getArgs();
+    if (cliArgs.length != 3)
+    {
+      _logger.error("Invalid arguments");
+      help();
+      System.exit(1);
+    }
+    int i = 0;
+    String resolverPath = RestLiToolsUtils.readArgFromFileIfNeeded(cliArgs[i++]);
+    String inputPath = cliArgs[i++];
+    String outputPath = cliArgs[i];
```

Comment:
For clarity, consider just using `0`, `1`, and `2` directly instead of incrementing `i`.